src/python/pants/core/goals/test.py

```diff
@@ -204,6 +224,127 @@ class TestFieldSet(FieldSet, metaclass=ABCMeta):
     __test__ = False
 
 
+_FieldSetT = TypeVar("_FieldSetT", bound=FieldSet)
+
+
+class Partitions(FrozenDict[Any, "tuple[_FieldSetT, ...]"]):
```

Comment:
oh I deleted my last comment. I misread the decorator. `@runtime_ignore_subscripts` was right.